PMK-550-Doc-Image-Processor_EN (#132)
aziontech/docs_en: image-processor/2021-01-14-index.md

The Azion **Image Processor** automates your workflow for processing images.

> Optimizing your images reduces their size without any noticeable loss of visual quality, which reduces the transfer time of your content and improves the user experience. This increases the fluidity of your pages and the speed of navigation, without you having to do practically anything.

Reviewer: ```suggestion
Processing your images reduces their size without any noticeable loss of visual quality, which reduces the transfer time of your content and improves the user experience. This increases the fluidity of your pages and the speed of navigation, without you having to do practically anything.
```

    \.(jpg|jpeg|gif|bmp|png)$
    ~~~

3. Choose the logical operator **Matches**, if you used the regular expression in the **Path** field;

4. In the **Behavior** field **Set Cache Settings**, select the preset used in Stage 2.

5. Also select **Optimize Images** in the **Behavior** field;

6. In all other sections, use the appropriate settings for your needs;

7. Click the **Save** button to save your rule.

> From this point on, images from the configured path will be automatically optimized. In addition, the **Image Processor** detects the browser's compatibility with the WEBP format and, when possible, will convert the image format automatically, bringing even more gains for you. BMP images will also be automatically converted to JPEG or WEBP, depending on the browser's compatibility.

Reviewer: ```suggestion
From this point on, images from the configured path will be automatically processed. In addition, the **Image Processor** detects the browser's compatibility with the WEBP format and, when possible, will convert the image format automatically, bringing even more gains for you. BMP images will also be automatically converted to JPEG or WEBP, depending on the browser's compatibility.
```

> Follow the *Bandwidth Saving* graph through **Real-Time Metrics** to monitor the traffic savings resulting from the optimization.

Reviewer: ```suggestion
Follow the *Bandwidth Saving* graph through **Real-Time Metrics** to monitor the traffic savings resulting from the processed images.
```

**Example:**

_https://yourdomain.com/image.jpg?ims=fit-in/400x400_ (maximum width of 400 pixels and maximum height of 400 pixels)

[![bulldog]({{ site.url }}/static/images/docs/image-optimization/bulldog-1280px.jpg?ims=fit-in/400x400)]({{ site.url }}/static/images/docs/image-optimization/bulldog-1280px.jpg?ims=fit-in/400x400)

---

## 4. Cropping the image {#crop}

The image cropping can be done by entering a start point (AxB) and an end point (CxD), as an argument in the Query String of the image URL:

~~~
ims=AxB:CxD
~~~

*   AxB: the starting point for the crop indicates the coordinates, in pixels, of the top left corner of the area to be cropped.
*   CxD: the end point for the crop indicates the coordinates, in pixels, form the lower right corner of the area to be cropped.

**Example:**

_http://yourdomain.com/image.jpg?ims=430x20:910x730_

[![bulldog]({{ site.url }}/static/images/docs/image-optimization/bulldog-1280px.jpg?ims=430x20:910x730)]({{ site.url }}/static/images/docs/image-optimization/bulldog-1280px.jpg?ims=430x20:910x730)

---

## 5. Changing image quality {#change-quality}

> The **Image Processor** optimizes your images, reducing file size and, therefore, the transfer time. The default quality value used is 85%, which provides optimization without any noticeable loss of visual quality.

Reviewer: ```suggestion
The **Image Processor** process your images, reducing file size and, therefore, the transfer time. The default quality value used is 85%, which provides optimization without any noticeable loss of visual quality.
```
